docs/howto/add_link.md

```diff
@@ -0,0 +1,336 @@
+  attributes :name
+  has_many :posts do
+    link(:self) do
+      include_data false
+      user_posts_path(object.id) # Rails url helper automatically available
+    end 
+  end
+end
+
+user = User.new(id: 1, name: "Kelly", posts: [ Post.new(id: 1) ])
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "relationships": {
+      "posts": {
+        "links": {
+          "self": "users/1/posts"
+        }
+      }
+    }
+  }
+}
+```
+
+## JSON adapter
+
+The `json` is not able to handle links defined by the `link` class method. However, you can either use the `meta`  member or define a `links` attribute on the serializer.
+
+### Via `meta`
+
+```ruby
+user = User.new(id: 1, name: "John")
+meta = {
+  links: {
+    self: "/users/#{user.id}",
+    posts: "/posts?user_id=#{user.id}"
+  }
+}
+resource = ActiveModelSerializer::SerializableResource.new(
+  user,
+  adapter: :json,
+  meta: meta
+)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "meta": {
+    "links": {
+      "self": "/users/1",
+      "posts": "/posts?user_id=1"
+    }
+  },
+  "user": {
+    "id": "1",
+    "name": "Example User",
+  }
+}
+```
+
+When using Rails, you can add links by providing the `meta` option to `render` method:
+
+```ruby
+class UserController < ActionController::Base
+  def show
+    user = User.new(id: 1, name: "John")
+     meta = {
+      links: {
+        self: "/users/#{user.id}",
+        posts: "/posts?user_id=#{user.id}"
+      }
+    }
+    render json: user, adapter: :json, meta: meta
+  end
+end
+```
+
+### Via an attribute
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :name, :links
+  attribute :links do
+    {
+      self: "/users/#{object.id}",
+      posts: "/posts?user_id=#{object.id}"
+    }
+  end
+end
+
+user = User.new(id: 1, name: "John")
+resource = ActiveModelSerializer::SerializableResource.new(user, adapter: :json)
+resource.to_json
+```
+ 
+This would result in:
+```json
+{
+  "user": {
+    "id": "1",
+    "name": "Example User",
+    "links": {
+      "self": "/users/1",
+      "posts": "/posts?user_id=1"
+    }
+  }
+}
+```
+
+If you want to use the Rails' url/path helpers such as `user_path`, you need to `include` the `ActiveModelSerializers::SerializationContext::UrlHelpers` module into your serializer:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  include ActiveModelSerializers::SerializationContext::UrlHelpers
+
+  attributes :id, :name, :links
+  attribute :links do
+    {
+      self: user_path(object.id),
+      posts: posts_path(user_id: object.id)
+    }
+  end
+end
+```
+
+## Attributes adapter
+
+The only way to provide pagination with the `attributes` adapter is to define a `links` attribute [as described for the JSON adapter above](#via-an-attribute).
```

Comment:
which honestly, is kind of dumb of us.  Why do we *forbid* having a meta attribute? Attributes adapter really needs to be reabsorbed into JSON, since it's really the domain of the serializer.